pymongo/server_description.py

```diff
@@ -69,6 +70,11 @@ def __init__(
         self._me = ismaster.me
         self._last_update_time = _time()
         self._error = error
+        # PYTHON-2433 Clear error traceback info.
```

Comment:
Any particular reason you are purging the traceback, etc. in this location as opposed to after emitting the heartbeat failed event in `Monitor._check_server`? Why preclude any exception on a ServerDescription from having a traceback?

Reply:
Yeah it is a bit odd to mutate the exception in the ServerDescription constructor. I moved this logic to the Monitor.